src/app/routes/homePage/getInitialData/index.ts

```diff
@@ -0,0 +1,79 @@
+/* eslint-disable @typescript-eslint/ban-ts-comment */
+import { Agent } from 'https';
+import Url from 'url-parse';
+import nodeLogger from '../../../lib/logger.node';
+import { BFF_FETCH_ERROR } from '../../../lib/logger.const';
+import fetchPageData from '../../utils/fetchPageData';
+import { Services } from '../../../models/types/global';
+import HOME_PAGE_CONFIG from './page-config';
+
+const logger = nodeLogger(__filename);
+
+type Envs = 'test' | 'live' | 'local' | 'undefined';
+
+const getEnvironment = (pathname: string) => {
+  if (pathname.includes('renderer_env=test')) {
+    return 'test';
+  }
+  if (pathname.includes('renderer_env=live')) {
+    return 'live';
+  }
+
+  return process.env.SIMORGH_APP_ENV as Envs;
+};
+
+type Props = {
+  getAgent: () => Promise<Agent>;
+  service: Services;
+  path: string;
+  pageType: 'homePage';
+};
+
+export default async ({ getAgent, service, path: pathname }: Props) => {
+  try {
+    const env = getEnvironment(pathname);
+    const isLocal = !env || env === 'local';
+
+    const agent = !isLocal ? await getAgent() : null;
+    const id = !isLocal ? HOME_PAGE_CONFIG[service][env] : null;
```

Comment:
Can we change these to be "positive" statements (so we don't need to translate double negatives 🤯 )

```suggestion
    const agent = isLocal ? null : await getAgent();
    const id = isLocal ? null : HOME_PAGE_CONFIG[service][env];
```

Reply:
Could try: 
`    // @ts-ignore
    const id = !isLocal ? HOME_PAGE_CONFIG[service][env] : null;`

or

`const id = !isLocal ? HOME_PAGE_CONFIG[service][env as keyof typeof HOME_PAGE_CONFIG[typeof service]]: null;`

Reply:
We've managed to get `HOME_PAGE_CONFIG[service][env]` working without having to use `@ts-ignore`! Or that very complicated / tricky `keyof typeof` functionality 🤯 But thank you for the suggestion @shayneahchoon!